### PR TITLE
Add unit tests for stream_reduce() - issue 100

### PR DESCRIPTION
**Pull request by AndresSanchezCuadrado**
unit test stream_reduce

### CODE REVIEW

File: unit_tests/stream_reduce.cpp [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+  }
+
+  void check_multiple_sink2() {
+    EXPECT_EQ(7, invocations_gen);
+    EXPECT_EQ(5, invocations_reduce);
+
+    EXPECT_EQ(35, this->out);
+  }
+
+/*
```
Comment (jdgarciauc3m) [inline unit_tests/stream_reduce.cpp]: Remove comments before merge

File: unit_tests/stream_reduce.cpp [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+
+// Test for execution policies defined in supported_executions.h
+TYPED_TEST_CASE(stream_reduce_test, executions);
+
+TYPED_TEST(stream_reduce_test, static_empty)
+{    
+
+  this->setup_empty();
+  grppi::stream_reduce(this->execution_,
+    [this]() {
```
Comment (jdgarciauc3m) [inline unit_tests/stream_reduce.cpp]: [this]() -> optional<int> {
  this->invocations++;
  return {};
}

**Review by jdgarciauc3m — COMMENTED** [review]
In genera the test cases need to be discussed in the next developers meeting.

File: unit_tests/stream_reduce.cpp (hunk 2 above)
Comment (jdgarciauc3m) [reply unit_tests/stream_reduce.cpp]: This has not been addressed.

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
Adress remaining comments.
Try to name test cases with meaningful names. If needed add some comment.

**Comment by AndresSanchezCuadrado** [comment]
It has not been fully tested. It has a dependency with another issue, as soon as it is solved we could test this one.

**Comment by jdgarciauc3m** [comment]
Please, document which issue we depend on.

**Comment by AndresSanchezCuadrado** [comment]
Depends on issue #127: Revise stream_reduce().

**Comment by jdgarciauc3m** [comment]
This branch does not compile.

**Review by jdgarciauc3m — APPROVED** [review]
OK

**Comment by jdgarciauc3m** [comment]
Solves issue #100

**Pull request merged**